### PR TITLE
Added changes from feedback call

### PR DESCRIPTION
**Pull request by sreeram-venkitesh**
- One-line PR description:

- Issue link:

- Other comments:

### CODE REVIEW

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 1]
```diff
@@ -1012,7 +1041,7 @@ This through this both in small and large cases, again with respect to the
 [supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
 -->
 
-Planning to benchmark this before beta.
+Since the CEL expressions are compiled and evaluated in the kube-apiserver, depending on the complexity of the CRDs and the expressions defined, we may see a non-negligible increase of CPU usage. We are planning to benchmark this before beta graduation.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md:1046]: :+1:

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 2]
```diff
@@ -843,6 +870,8 @@ checking if there are objects with field X set) may be a last resort. Avoid
 logs or events for this purpose.
 -->
 
+The cluster admin can check if the CRDAdditionalPrinterColumnCEL feature gate is turned on. If yes, the admin can check if any CRD has additionalPrinterColumns which are defined with `expression` instead of `jsonPath`.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: remove the instead of `jsonPath` part?

```suggestion
The cluster admin can check if the CRDAdditionalPrinterColumnCEL feature gate is turned on.
If yes, the admin can further check if any CRD has any columns defined under `additionalPrinterColumns` section which are using the new `expression` field.
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 3]
```diff
@@ -816,7 +843,7 @@ Longer term, we may want to require automated upgrade/rollback tests, but we
 are missing a bunch of machinery and tooling and can't do that now.
 -->
 
-No, a Kubernetes upgrade/downgrade operation is not expected to affect this feature.
+We're planning to test upgrade-> downgrade -> upgrades before graduating to beta.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md:848]: :+1:

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 4]
```diff
@@ -779,7 +806,7 @@ You can take a look at one potential example of such test in:
 https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
 -->
 
-We will add an integration test to ensure that the feature is disabled when the feature gate is off.
+We have unit and integration tests to make sure that the feature enablement and disablement works as intended.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: I thought we can still add "will have" instead of "have" :sweat_smile: 

```suggestion
We will have unit and integration tests to make sure that the feature enablement and disablement works as intended.
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 5]
```diff
@@ -758,11 +785,11 @@ feature.
...
 -->
 
-Yes, the feauter can be disabled once it has been enabled. Enabling this feature would let users define additionalPrinterColumns in their custom resource definitions with CEL expressions instead of JSONPath. Existing JSONPath support is untouched. If the feature is disabled, the existing additionalPrinterColumns with JSONPaths would work as expected. Resources with CEL expressions in their additionalPrinterColumn definition would be invalid if we disable the feature however.
+Yes, if the feature is disabled after being used, the existing additionalPrinterColumns with JSONPaths would work as expected. Existing resources with CEL expressions in their additionalPrinterColumn definition would be ignored and those columns will not be printed if the feature is disabled.
 
 ###### What happens if we reenable the feature if it was previously rolled back?
 
-Nothing. CRDs which had failed validation previously might now succeed if the CEL expression is valid.
+CRDs which had failed validation previously might now succeed if the CEL expression is valid. Existing CRDs additionalPrinterColumns defined with CEL expression would start working again after the feature has been reenabled.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md:794]: :+1:

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 6]
```diff
@@ -758,11 +785,11 @@ feature.
 NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
 -->
 
-Yes, the feauter can be disabled once it has been enabled. Enabling this feature would let users define additionalPrinterColumns in their custom resource definitions with CEL expressions instead of JSONPath. Existing JSONPath support is untouched. If the feature is disabled, the existing additionalPrinterColumns with JSONPaths would work as expected. Resources with CEL expressions in their additionalPrinterColumn definition would be invalid if we disable the feature however.
+Yes, if the feature is disabled after being used, the existing additionalPrinterColumns with JSONPaths would work as expected. Existing resources with CEL expressions in their additionalPrinterColumn definition would be ignored and those columns will not be printed if the feature is disabled.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: minor typo?

```suggestion
Yes, if the feature is disabled after being used, the existing additionalPrinterColumns with JSONPath would work as expected. Existing resources with CEL expressions in their additionalPrinterColumn definition would be ignored and those columns will not be printed if the feature is disabled.
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 7]
```diff
@@ -682,7 +707,9 @@ enhancement:
   CRI or CNI may require updating that component before the kubelet.
 -->
 
-N/A
+This feature is implemented in the kube-apiserver component, skew with other kubernetes components do not require coordinated behavior.
+
+Clients should ensure the kube-apiserver is fully rolled out before using the feature.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md:714]: :+1:

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 8]
```diff
@@ -663,9 +684,13 @@ enhancement:
...
-No change in how users upgrade/downgrade their clusters. This feature may remove
-complexity by removing risk that a tightened validation on Kubernetes' part
-does not break workflow.
+<!-- No change in how users upgrade/downgrade their clusters. This feature may remove -->
+<!-- complexity by removing risk that a tightened validation on Kubernetes' part -->
+<!-- does not break workflow. -->
+
+No changes are required for a cluster to make an upgrade and maintain existing behavior. There is new API that does not effect the cluster during upgrade. It only has effects if it is used after the upgrade.
+
+If a cluster is downgraded, no changes are required. The cluster continues to work as expected since the alpha version will have functionality compatible with beta and stable release, any downgrade will be to a version that also contains the feature.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: This wording still doesn't make sense to me sadly :(

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 9]
```diff
@@ -640,11 +659,13 @@ in back-to-back releases.
...
 - Gather feedback from developers and surveys
 - Add e2e tests
-- Add appropriate metrics - for additionalPrinterColumn usage and CEL cost usage
+- Add appropriate metrics for additionalPrinterColumns usage and CEL cost usage
 - More benchmarking to compare JSONPath and CEL execution and modify CEL cost if needed
 
 #### GA
 
+- N examples of real-world usage
+- N installs
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: N installs

How we will come up with stats for this?

N examples of real-world usage makes proper sense to me since we already know few areas which will immediately start using this feature once available.

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 10]
```diff
@@ -564,7 +582,8 @@ https://storage.googleapis.com/k8s-triage/index.html
 We expect no non-infra related flakes in the last month as a GA graduation criteria.
 -->
 
-- <test>: <link to test coverage>
+<!-- - <test>: <link to test coverage> -->
+We will test the edge cases mostly in integration test and unit test. If needed, we can add e2e tests before beta graduation.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]: what edge cases? and `before beta graduation` means not alpha right? it means when we target beta graduation?

```suggestion
We will test the edge cases mostly in integration test and unit test. If needed, we can add e2e tests while planning for beta graduation.
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 11]
```diff
@@ -518,15 +530,21 @@ extending the production code to implement this enhancement.
 
 Alpha:
 
+TODO: Mention which files we'll edit
+
+staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation_test.go
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]:
```suggestion
[staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation_test.go](https://github.com/kubernetes/kubernetes/blob/e54719bb6674fac228671e0786d19c2cf27b08a3/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation_test.go)
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 12]
```diff
@@ -518,15 +530,21 @@ extending the production code to implement this enhancement.
...
 - Test that validation passes when we create an additionalPrinterColumn with an expression field with valid CEL expression
 - Test that validation fails when we create an additionalPrinterColumn with an expression field with an invalid CEL expression
 - Test that existing behaviour of jsonPath is not altered when creating CRDs with only jsonPath additionalPrinterColumns
 - Test that validation fails when we create an additionalPrinterColumn with both jsonPath and expression fields
 - Test that validation passes when we create multiple additionalPrinterColumns with both jsonPath and expression fields
 - Test that validation fails when we try to create an additionalPrinterColumn with expression field when the feature gate is turned off
-- Verify that CEL compilation errors are caught at the validation phase
-- Verify that TableConvertor is getting created for the CRD with both jsonPath and expression columns
+
+staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]:
```suggestion
[staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go](https://github.com/kubernetes/kubernetes/blob/bc302fa4144d21a338683cd83701661f97be4aba/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go)
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 13]
```diff
@@ -545,7 +563,7 @@ For Alpha, describe what tests will be added to ensure proper quality of the enh
 For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
 https://storage.googleapis.com/k8s-triage/index.html
 -->
-
+test/integration/apiserver/crd_additional_printer_columns_test.go
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md]:
```suggestion
[test/integration/apiserver/crd_additional_printer_columns_test.go](https://github.com/kubernetes/kubernetes/tree/bc302fa4144d21a338683cd83701661f97be4aba/test/integration/apiserver)
```

File: keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md [hunk 14]
```diff
@@ -433,43 +443,45 @@ func (c additionalPrinterColumnCelCompilationResult) FindResults(data interface{
...
 }
 ```
 
 With this, the current `ConvertToTable` method can support additionalPrinterColumns without any changes.
 
-### Cost Calculation and benchmarking performance with jsonPath
+### Cost Calculation and benchmarking performance with JSONPath
 
-A big part of the discussions for our proposal was the CEL cost limits since this is the first time CEL is added to the read path. As part of this we've done some benchmarking of the time it takes to parse and compile equivalent jsonPath and CEL expressions.
+A big part of the discussions for our proposal was the CEL cost limits since this is the first time CEL is added to the read path. As part of this we've done some benchmarking of the time it takes to parse and compile equivalent JSONPath and CEL expressions.
```
Comment (Priyankasaggu11929) [inline keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md:478]: :+1:
